fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java

```diff
@@ -869,6 +869,11 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
                 childTypes[i] = children.get(i).type;
             }
 
+        } else if (fnName.getFunction().equalsIgnoreCase("if")) {
+            Type[] childTypes = collectChildReturnTypes();
+            Type assignmentCompatibleType = ScalarType.getAssignmentCompatibleType(childTypes[1], childTypes[2], true);
+            childTypes[1] = assignmentCompatibleType;
+            childTypes[2] = assignmentCompatibleType;
             fn = getBuiltinFunction(analyzer, fnName.getFunction(), childTypes,
```

Comment:
After this change, FunctionSet.WINDOW_FUNNEL will not call getBuiltinFunction any more. 

It seams FunctionSet.WINDOW_FUNNEL will work not correctly.

Reply:
My problem, I will fix it.